src/backend/distributed/executor/insert_select_executor.c

```diff
@@ -770,6 +770,18 @@ AddInsertSelectCasts(List *insertTargetList, List *selectTargetList,
 			if (selectEntry->ressortgroupref != 0)
 			{
 				selectEntry->resjunk = true;
+
+				/*
+				 * This entry might still end up in the SELECT output list, so
+				 * rename it to avoid ambiguity.
+				 *
+				 * See https://github.com/citusdata/citus/pull/3470.
+				 */
+				resnameString = makeStringInfo();
+				appendStringInfo(resnameString, "discarded_target_item_%d",
```

Comment:
Does this appear in explain output? If so it might make sense to add the original column name in there somewhere.

Reply:
In the case I tried, postgres replaces these target entries with the expression they represent in the explain output. So no need to include it. I will look into it more tomorrow.